actions/build-and-push-image-digests/action.yaml

```diff
@@ -0,0 +1,147 @@
+name: Build and push image digests to GitHub
+description: Composite action to build and push image digests to GitHub
+
+inputs:
+  registry:
+    description: |
+      Google Artifact Registry to store docker images in.
+    default: "us-docker.pkg.dev"
+  tag:
+    description: |
+      Docker images tag to be pushed.
+    required: true
+  context:
+    description: |
+      Path to the Docker build context.
+    default: "."
+  environment:
+    description: |
+      Environment for pushing artifacts (can be either dev or prod).
+    default: dev
+  image_name:
+    description: |
+      Name of the image to be pushed to GAR.
+    required: true
+  build-args:
+    description: |
+      List of arguments necessary for the Docker image to be built.
+    default: ""
+  file:
+    description: |
+      The dockerfile to use.
+    required: false
+  platform:
+    description: |
+      Platforms to build the image for
+    required: true
+  cache-from:
+    description: |
+      Where cache should be fetched from
+    required: false
+    default: "type=gha"
+  cache-to:
+    description: |
+      Where cache should be stored to
+    required: false
+    default: "type=gha,mode=max"
+  ssh:
+    description: |
+      List of SSH agent socket or keys to expose to the build
+  build-contexts:
+    description: |
+      List of additional build contexts (e.g., name=path)
+    required: false
+  docker-buildx-driver:
+    description: |
+      The driver to use for Docker Buildx
+    required: false
+    default: "docker-container"
+
+outputs:
+  full_image_name:
```

Comment:
Dirty but for now re-calculating the whole image name for GAR is difficult, we can get away with it using a simple output.